lib/galaxy/managers/markdown_util.py

```diff
@@ -122,6 +123,14 @@ def _remap(container, line):
                 assert object_id is not None
                 hda = hda_manager.get_accessible(object_id, trans.user)
                 rval = self.handle_dataset_info(line, hda)
+            elif container == "history_dataset_type":
+                assert object_id is not None
+                hda = hda_manager.get_accessible(object_id, trans.user)
+                rval = self.handle_dataset_type(line, hda)
+            elif container == "history_dataset_name":
+                assert object_id is not None
+                hda = hda_manager.get_accessible(object_id, trans.user)
+                rval = self.handle_dataset_name(line, hda)
```

Comment:
we really should put these things in a separate function and run it against array of keywords. Because we have pretty much 100 identical lines.

Reply:
It is is sort of common in visitor patterns but you're probably right - this could be made more precise for sure.